src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs

```diff
@@ -0,0 +1,261 @@
+using static Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    public class BlueprintDefinitionCmdletBase : BlueprintCmdletBase
+    {
+
+        /// <summary>
+        /// Get management group ancestors for a given subscription
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <returns></returns>
+        protected List<string> GetManagementGroupAncestorsForSubscription(string subscriptionId)
+        {
+            List<string> managementGroupAncestors = new List<string>();
+
+            if (subscriptionId != null)
+            {
+                string result = GetManagementGroupAncestorsAsync(subscriptionId).GetAwaiter().GetResult();
+                var resultJObjects = JObject.Parse(result);
+                var managementGroupAncestorsObjects = resultJObjects["managementGroupAncestors"].Children().ToList();
+
+                foreach (var mgObject in managementGroupAncestorsObjects)
+                {
+                    managementGroupAncestors.Add(mgObject.ToString());
+                }
+            }
+            return managementGroupAncestors;
+        }
+
+        /// <summary>
+        /// Task for get management group ancestors
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <returns></returns>
+        private async Task<string> GetManagementGroupAncestorsAsync(string subscriptionId)
+        {
+            var url = string.Format(BlueprintConstants.MgAncestorsRequestUrlTemplate, subscriptionId);
+            var httpRequest = new HttpRequestMessage();
+            httpRequest.Method = new HttpMethod("GET");
+            httpRequest.RequestUri = new Uri(url);
+
+            HttpResponseMessage httpResponse = null;
+            string responseContent = null;
+
+            try
+            {
+                using (HttpClient client = new HttpClient())
+                {
+                    await ClientCredentials.ProcessHttpRequestAsync(httpRequest, new CancellationToken(false)).ConfigureAwait(false);
+                    httpResponse = await client.SendAsync(httpRequest, new CancellationToken(false));
+
+                    HttpStatusCode statusCode = httpResponse.StatusCode;
+                    // If we can't find the given subscription in the tenant, show error message.
+                    if (statusCode == HttpStatusCode.NotFound)
+                    {
+                        CloudException cex = new CloudException(string.Format("Subscription Id '{0}' could not be found in current tenant.", subscriptionId));
+                        throw cex;
+                    }
+
+                    responseContent = httpResponse.EnsureSuccessStatusCode().Content.ReadAsStringAsync().Result;
+                }
+                return responseContent;
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
```

Comment:
@filizt we shouldn't use `Console.WriteLine()` here -- why is this needed if we're throwing the exception?

Reply:
Left over from debugging. Removing.